lib/cli.js

```diff
@@ -18,27 +18,41 @@ var getGitFolder = require('./getGitFolder');
 // hacky start if not run by mocha :-D
 // istanbul ignore next
 if (process.argv.join('').indexOf('mocha') === -1) {
+  var commitMsgFileOrText = process.argv[2] || getGitFolder() + '/COMMIT_EDITMSG';
 
-  var commitMsgFile = process.argv[2] || getGitFolder() + '/COMMIT_EDITMSG';
-  var incorrectLogFile = commitMsgFile.replace('COMMIT_EDITMSG', 'logs/incorrect-commit-msgs');
+  var bufferToString = function (buffer) {
+    return hasToString(buffer) && buffer.toString();
+  };
 
-  var hasToString = function hasToString(x) {
+  var hasToString = function (x) {
     return x && typeof x.toString === 'function';
   };
 
-  fs.readFile(commitMsgFile, function(err, buffer) {
-    var msg = getCommitMessage(buffer);
-
+  var validate = function (msg, isFile) {
     if (!validateMessage(msg)) {
+      var incorrectLogFile = (
+        isFile
+        ? commitMsgFileOrText.replace('COMMIT_EDITMSG', 'logs/incorrect-commit-msgs')
+        : (getGitFolder() + '/logs/incorrect-commit-msgs')
+      );
+
       fs.appendFile(incorrectLogFile, msg + '\n', function() {
         process.exit(1);
       });
     } else {
       process.exit(0);
     }
+  };
 
-    function getCommitMessage(buffer) {
-      return hasToString(buffer) && buffer.toString();
+  fs.stat(commitMsgFileOrText, function(err, stat) {
```

Comment:
I assume you use `fs.stat` here to check existence of the file. This is not recommended by [node docs](https://nodejs.org/dist/latest-v6.x/docs/api/fs.html#fs_fs_stat_path_callback). You could use `fs.readFile` and if there is an `ENOENT` error execute `validate(commitMsgFileOrText, false);`. Would this work?